src/Core/Kernel.php

```diff
@@ -346,7 +346,7 @@ private function instantiatePlugins(array $plugins): void
             }
 
             /** @var Plugin $plugin */
-            $plugin = new $className((bool) $pluginData['active']);
+            $plugin = new $className((bool) $pluginData['active'], $this->getProjectDir() . $pluginData['path']);
```

Comment:
please consider that the path to the plugin root and the path to the plugin base class could be different. so this is not correct solution for the symlink use case :wink:

Reply:
Supporting symlinks while still resolving all resources relative to the Plugin's base class in a clean way is likely impossible, since the path of a class in PHP is unfortunately always canonicalized (which is the root cause of all the symlinking problems). I have a somewhat hacky solution on this branch: https://github.com/shopware/platform/compare/master...VIISON:pickware/experimental-compute-non-canonicalized-plugin-class-path#diff-b3226395e77d614094c66a2d3202f556R67

But as you can see, that's not pretty either.

I've also discovered that resolving resources relative to the plugin base class breaks layouts that look like this:

```
.
├── Resources
│   └── administration
│       └── main.js
└── src
    └── MyPlugin.php
```

Because apparently, overriding `\Shopware\Core\Framework\Bundle::getAdministrationEntryPath` with `return '../Resources/administration'` doesn't work either.

Also, with the relative-to-base-class model, the values for the `plugin.path` column and the value of  `Plugin->getPath()` differ, which feels kind of inconsistent.

I think resolving resources relative to the plugin root is the more logical and consistent option and causes less problems than resolving relative to the base class. If resolving relative to the base class is required, it should be trivial to override the `get*Path()` methods in the plugin to achieve this - the opposite case unfortunately doesn't work.

I've also already explained the non-standard layout problem shown by the filesystem tree above to @janbuecker.

All in all, I'd ask you to reconsider dropping the relative-to-base-class resolution in favor of resolving resources relative to the plugin root.

If that's not a workable solution for you, we'd have to use the hacky workaround on the https://github.com/shopware/platform/compare/master...VIISON:pickware/experimental-compute-non-canonicalized-plugin-class-path branch or drop symlink support, which would be pretty inconvenient for plugin developers. Also, I have no idea on how to make the `/Resources` case described above work correctly with that approach.

Reply:
Heey, you know my pull request? I have changed the plugin logic so plugin != bundle. so can have bundle in subdir.
Also i testet it with symlink. also the no bootstrap option works so. projectdir + path